lib/vagrant_cloud/box.rb

```diff
@@ -37,7 +37,13 @@ def initialize(organization:, **opts)
     # @note This will delete the box, and all versions
     def delete
       if exist?
-        organization.account.client.box_delete(username: username, name: name)
+        organization.account.client.box_delete(
+          username: username,
+          name: name
+        )
+        b = organization.boxes.dup
```

Comment:
(question here more for my understanding of whats going on) Why do the boxes get a `dup` and then deleted before being cleared out?

Reply:
When the information for an instance is loaded, the data is frozen so that it is immutable. This is done here: https://github.com/hashicorp/vagrant_cloud/blob/main/lib/vagrant_cloud/data.rb#L173

When making changes to the instance, changes are kept in the `dirty` dataset which allows us to easily check if an instance has been modified and requires a save. For the `add_` methods, we want to just inject our new value in without marking the instance dirty or requiring a reloading of the data. To do that we just make a copy of the frozen array (so that it is no longer frozen and we can modify it) and add our new instance to it. Then it's set back into the data as clean so the instance doesn't show as requiring to be saved.

Reply:
thanks for the explanation!